providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/grpc.py

```diff
@@ -146,15 +148,37 @@ def listen(self) -> None:
                 )
 
             self.connected = False
+            self.handle_error(retry_counter, retry_delay)
+
+            retry_delay = self.handle_retry(retry_counter, retry_delay)
+
+            retry_counter = retry_counter + 1
```

Comment:
Just for my understanding. The `retry_counter `seems to be incremented (and the error and retry handling exectued) in every loop iteration, even if the connection succeeds, or am I missing something?

Reply:
You are right. `handle_error` is a bad name for what I am actually doing. Two things are hidden behind it. 

1. if we hit our grace_attempts with the `error_counter` we emit an error event
2. if our immediate reconnect did not work, we're emitting a stale event on the first time.